addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -600,6 +606,48 @@ var SnippetEditor = Widget.extend({
             },
         });
     },
+    /**
+     *
+     * @param {*} ev
+     * @param {*} ui
+     */
+    _onDragAndDropDrag: function (ev, ui) {
+        let $parent = getClosestScrollableParent(this.$target.parent());
+        $parent = $('body');
+        let topOffset;
+        let scroll;
+
+        // OSU TODO : replace this code by something that find the closest scrollable parent
+        if ($parent.is('body')) {
+            $parent = $(window);
+            topOffset = window.pageYOffset;
+            scroll = window.scrollY;
+        } else {
+            topOffset = $parent.offset().top;
+        }
+
+        // 1. Find the closest srollable parent
+        // 2. Get its offset and height values
+        // 3. Depending on the mouse position, create an interval to go up or down
+        // -> Check on internet how to properly delete an interval and avoid to create 2 at the same tile
+        // 4. Test it, and do the same for the snippet dragger.
+        if (topOffset + 70 >= ui.position.top) {
+            if (!this.scrollInterval) {
+                this.scrollInterval = setInterval(() => {
+                    $parent.scrollTop($parent.scrollTop() - 5, 100);
+                });
+            }
+        } else if (topOffset + $parent.innerHeight() - 70 <= ui.position.top && this.scrollHeightLimit > scroll + $parent.innerHeight()) {
+            if (!this.scrollInterval) {
+                this.scrollInterval = setInterval(() => {
+                    $parent.scrollTop($parent.scrollTop() + 5, 100);
+                });
+            }
+        } else {
+            clearInterval(this.scrollInterval);
+            this.scrollInterval = undefined;
+        }
```

Comment:
Why... so here on each drag you start a setInterval with not even a delay... is the goal trying to make your computer explode or something like that?

If you want to try something this way (which I don't know if it will work), why not starting your setInterval in start, stoping it in stop and updating some position/whatever info in drag?

Reply:
I start a setInterval only if there isn't any that already exist, not on each drag. And I delete it when I'm no more in the area which allow to scroll down or up

the scroll properties of draggable specify in the doc that the window auto scroll, but I looked on internet if I was doing something wrong, but apparently not, it seems to not be a real auto scroll, if you stop moving your mouse, it stop scrolling (when you're at the top or the bottom of the page). So, to have something smooth I guess I had to figure out a way myself, didn't find any real other easy way on internet.

Reply:
Ok did not see that `if (!this.scrollInterval) {` check... You still cannot use a setInterval with no delay. Here, during the autoscroll there probably are thousands of calls to `$parent.scrollTop(` being made... with a delay of 100ms for each anyway. I tested your branch, the behavior is the exact same as the current master to me. Not smooth and I still have to move the cursor to be able to drag...

Worth a try:
- start: start a requestAnimationFrame loop
- stop: stop the animation frame loop
- in the animation frame loop: increase/decrease the scroll position depending on a scrollspeed
- drag: compute a "scroll speed" depending on the cursor position:

```
speed depending on cursor position:
--------------------------------------------
high negative speed
medium negative speed
low negative speed
speed = 0
speed = 0
speed = 0
low positive speed
medium positive speed
high positive speed
--------------------------------------------
```